hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBTable.java

```diff
@@ -155,6 +155,32 @@ public boolean isExist(byte[] key) throws IOException {
     }
   }
 
+  @Override
+  public byte[] getIfExist(byte[] key) throws IOException {
+    try {
+      // RocksDB#keyMayExist
+      // If the key definitely does not exist in the database, then this
+      // method returns false, else true.
+      rdbMetrics.incNumDBKeyGetIfExistChecks();
+      StringBuilder outValue = new StringBuilder();
+      boolean keyMayExist = db.keyMayExist(handle, key, outValue);
+      if (keyMayExist) {
+        // Not using out value from string builder, as that is causing
+        // IllegalArgumentException during protobuf parsing.
```

Comment:
whats the reason for the illegal arguement exception ? Do we have a stack trace?

Reply:
Because value returned from StringBuilder then to a string, which is converted to bytes[] and then later if it is used in parseFrom byte[] data.
```
java.lang.IllegalArgumentException: Can't encode the the raw data from the byte array
	at org.apache.hadoop.ozone.om.codec.OmKeyInfoCodec.fromPersistedFormat(OmKeyInfoCodec.java:48)
	at org.apache.hadoop.ozone.om.codec.OmKeyInfoCodec.fromPersistedFormat(OmKeyInfoCodec.java:31)
	at org.apache.hadoop.hdds.utils.db.CodecRegistry.asObject(CodecRegistry.java:55)
	at org.apache.hadoop.hdds.utils.db.TypedTable.getFromTableIfExist(TypedTable.java:207)
	at org.apache.hadoop.hdds.utils.db.TypedTable.getIfExist(TypedTable.java:194)
	at org.apache.hadoop.ozone.om.request.key.OMKeyCreateRequest.validateAndUpdateCache(OMKeyCreateRequest.java:196)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerRequestHandler.handleWriteRequest(OzoneManagerRequestHandler.java:230)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerProtocolServerSideTranslatorPB.submitRequestDirectlyToOM(OzoneManagerProtocolServerSideTranslatorPB.java:210)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerProtocolServerSideTranslatorPB.processRequest(OzoneManagerProtocolServerSideTranslatorPB.java:130)
	at org.apache.hadoop.hdds.server.OzoneProtocolMessageDispatcher.processRequest(OzoneProtocolMessageDispatcher.java:72)
	at org.apache.hadoop.ozone.protocolPB.OzoneManagerProtocolServerSideTranslatorPB.submitRequest(OzoneManagerProtocolServerSideTranslatorPB.java:98)
	at org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos$OzoneManagerService$2.callBlockingMethod(OzoneManagerProtocolProtos.java)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:524)
	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1025)
	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:876)
	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:822)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)




```

Reply:
Does this mean we cannot use the outValue that we pass in? If 'keyMayExist' returns true, and the value is indeed present in the block cache, I believe the outValue will have the data.

Reply:
Yes, using that is causing the issue.

Reply:
@Override
public byte[] get(byte[] key) throws IOException {
  try {
    // RocksDB#keyMayExist
    // If the key definitely does not exist in the database, then this
    // method returns false, else true.
    rdbMetrics.incNumDBKeyIfExistChecks();
    StringBuilder outValue = new StringBuilder();
    boolean keyMayExist = db.keyMayExist(handle, key, outValue);
    if (keyMayExist) {
      byte[] val;
      if (outValue.length() > 0) {
        val = outValue.toString().getBytes(UTF_8);
      } else {
        val = db.get(handle, key);
      }
      if (val != null) {
        rdbMetrics.incNumDBKeyIfExistMisses();
      }
      return val;
    }
    return null;
  } catch (RocksDBException e) {
    throw toIOException(
        "Error in accessing DB. ", e);
  }
}

I tried with the above code, getting IllegalException during parsing. So, for now, removed not to use outVal.